File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        int dp=0;
        
        for(int i=1;i<nums.size();i++){
            if(nums[dp]!=nums[i]){
                nums[++dp]=nums[i];
            }
        }
        dp++;
        
        return dp;
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Set<Integer> st = new LinkedHashSet<>();
        
        for(int i=0;i<nums.length;i++){
            st.add(nums[i]);
        }
        
        int i=0;
        for(int val:st){
            nums[i]=val;
            i++;
        }
        
        return st.size();
    }
}